### DISCUSSION TITLE
Dampen (debounce) request to revalidate individual page?

### DISCUSSION THREAD

**Discussion by vnugent** [vercel/next.js#36839] [Help]
Being able to request Next to regenerate a page on-demand is a huge benefit.  How would you go about 'dampening' or limiting the number of consecutive requests?  Is it possible to set `revalidate: <time>` and manually trigger a re-generation?  Something like, "regenerate this page whenever I request, but don't do it more than once every 5 minutes"?
 
https://nextjs.org/blog/next-12-1#on-demand-incremental-static-regeneration-beta

**Comment by vadymshymko** [comment]
Hello!

>Is it possible to set revalidate: <time> and manually trigger a re-generation? Something like, "regenerate this page whenever I request, but don't do it more than once every 5 minutes"?

Yes, it is possible) You can use [on demand revalidation](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation-beta)

All you need to do is create an api route that will call unstable_revalidate when accessed, and set a value for revalidate in getStaticProps
